packages/connect/src/api/ethereumGetAddress.ts

```diff
@@ -57,26 +61,50 @@ export default class EthereumGetAddress extends AbstractMethod<'ethereumGetAddre
             };
         });
 
-        // set info
+        const useEventListener =
+            payload.useEventListener &&
+            this.params.length === 1 &&
+            typeof this.params[0].address === 'string' &&
+            this.params[0].show_display;
+        this.confirmed = useEventListener;
+        this.useUi = !useEventListener;
+    }
+
+    async initAsync(): Promise<void> {
+        for (let i = 0; i < this.params.length; i++) {
+            if (this.params[i].network) {
+                // network was already set from 'well-known' definition in init method.
+                continue;
+            }
+            const slip44 = getSlip44ByPath(this.params[0].address_n);
+
+            const definitions = await getEthereumDefinitions({
+                slip44,
+            });
+
+            const decoded = decodeEthereumDefinition(definitions);
+
+            if (decoded.network) {
+                this.params[i].network = ethereumNetworkInfoFromDefinition(decoded.network);
+            }
+            if (definitions.encoded_network) {
+                this.params[i].encoded_network = definitions.encoded_network;
+            }
+        }
+    }
```

Comment:
It might make it more readable creating a variable for this.params[i] at the beginning like:

```suggestion
    async initAsync(): Promise<void> {
        for (let i = 0; i < this.params.length; i++) {
            const param = this.params[i];
            if (param.network) {
                // network was already set from 'well-known' definition in init method.
                continue;
            }
            const slip44 = getSlip44ByPath(this.params[0].address_n); // Maybe also here `param`, but I am not sure if it was intended.

            const definitions = await getEthereumDefinitions({
                slip44,
            });

            const decoded = decodeEthereumDefinition(definitions);

            if (decoded.network) {
                param.network = ethereumNetworkInfoFromDefinition(decoded.network);
            }
            if (definitions.encoded_network) {
                param.encoded_network = definitions.encoded_network;
            }
        }
    }
```

Reply:
well, maybe yes, bu I am assigning results to this.params[i] again in each iteration, which would make this on the other hand less readable I would say. One needs to keep track of references in his head

Reply:
Fair enough :)